### PR TITLE
Add : Support e2e( under 'concurrency' mode) outputting the summary of 'verify' in YAML format

### PR DESCRIPTION
**Pull request by chunriyeqiongsaigao**
1. Support e2e( in 'concurrency' mode) outputting the result of 'verify' in YAML format. 
2. Fix some nits.
3. Refactor the directory structure under /test/e2e.
4. Add congifuration files of concurrency mode to /test/e2e/.

After this PR is merged, the e2e( released version) will be able to test e2e( dev version) in 'concurrency' mode. In the next PR, the commented section in /test/e2e/e2e.yaml will be uncommented.

### CODE REVIEW

File: test/e2e/concurrency/fail-fast/internal/verify.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    # the interval between two attempts, e.g. 10s, 1m.
+    interval: 1s
+
+  # when a case fails, whether to stop verifying other cases. This property defaults to true.
+  fail-fast: true
+  # Whether to verify cases concurrently. This property defaults to false.
+  concurrency: true
+
+
+  cases:
```
Comment (fgksgf) [inline test/e2e/concurrency/fail-fast/internal/verify.yaml:32]: I think 10 cases are enough.

File: test/e2e/concurrency/non-fail-fast/internal/verify.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    # the interval between two attempts, e.g. 10s, 1m.
+    interval: 1s
+
+  # when a case fails, whether to stop verifying other cases. This property defaults to true.
+  fail-fast: false
+  # Whether to verify cases concurrently. This property defaults to false.
+  concurrency: true
+
+
+  cases:
```
Comment (fgksgf) [inline test/e2e/concurrency/non-fail-fast/internal/verify.yaml:32]: Ditto.

File: test/e2e/non-concurrency/fail-fast/expected.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  - passed-case-2
+  - passed-case-3
+  - passed-case-4
+  - passed-case-5
+failed:
+  - failed-case-6
+skipped:
+  - passed-case-7
+  - failed-case-8
+  - failed-case-9
```
Comment (fgksgf) [inline test/e2e/non-concurrency/fail-fast/expected.yaml:29]: It looks weird, can we rename these cases, like `case-X`

File: test/e2e/concurrency/fail-fast/expected.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- {{ if hasPrefix . "passed" }}{{.}}{{ end }}
+{{end}}
+failed:
+{{range .failed}}
+- {{ if hasPrefix . "failed" }}{{.}}{{ end }}
+{{end}}
+skipped:
+{{range .skipped}}
+- {{.}}
+{{end}}
```
Comment (fgksgf) [inline test/e2e/concurrency/fail-fast/expected.yaml:29]: What's the point of this?

**Review by fgksgf — COMMENTED** [review]
Basically LGTM, please polish it.

File: test/e2e/concurrency/fail-fast/expected.yaml (hunk 4 above)
Comment (chunriyeqiongsaigao) [reply test/e2e/concurrency/fail-fast/expected.yaml:29]: The "point" is the element of the "skipped" array in 'range' syntax.

**Review by fgksgf — APPROVED** [review]

**Pull request merged**